src/opm/parser/eclipse/EclipseState/SimulationConfig/RockConfig.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+  Copyright 2020 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <opm/parser/eclipse/Parser/ParserKeywords/R.hpp>
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/EclipseState/SimulationConfig/RockConfig.hpp>
+#include <opm/parser/eclipse/EclipseState/Grid/FieldPropsManager.hpp>
+
+namespace Opm {
+
+namespace {
+
+std::string num_prop(const std::string& prop_name) {
+    if (prop_name == "PVTNUM")
+        return "PVTNUM";
+
+    if (prop_name == "SATNUM")
+        return "SATNUM";
+
+    if (prop_name == "ROCKNUM")
+        return "ROCKNUM";
+
+    throw std::invalid_argument("The rocknum propertype: " + prop_name + " is not valid");
+}
+
+
+RockConfig::Hysteresis hysteris(const std::string& s) {
```

Comment:
Do you want to rename this function before we merge?

Reply:
Yes - done.